coretex/cli/commands/project.py

```diff
@@ -20,47 +20,106 @@
 import click
 
 from ..modules import ui, project_utils, utils, user
-from ...entities import Project
-from ...networking import NetworkRequestError
-from ...configuration import loadConfig, saveConfig
+from ..modules.ui import clickPrompt, errorEcho, successEcho, progressEcho
+from ...entities import Project, ProjectType, ProjectVisibility
+from ...networking import NetworkRequestError, EntityNotCreated, networkManager, RequestFailedError
+from ...configuration import loadConfig
 
 
-def selectProject(projectId: int) -> None:
+@click.command()
+@click.option("--project", "-p", type = str, help = "Project name")
+@click.option("--type", "-t", type = int, help = "Project type")
+@click.option("--description", "-d", type = str, help = "Project description")
+def create(name: Optional[str], type: Optional[int], description: Optional[str], visibility: Optional[int]) -> None:
+    if name is None:
+        name = clickPrompt("Please enter name of the project you want to create:", type = str)
+
+    if type is None:
+        type = project_utils.selectProjectType()
+
+    if description is None:
+        description = clickPrompt("Please enter your project's description:", type = str)
+
+    try:
+        project = Project.createProject(name, ProjectType(type), description)
+
+        if project is not None:
+            ui.successEcho(f"Project \"{name}\" created successfully.")
+            selectNewProject = clickPrompt("Do you want to select new project as default? (Y/n)", type = bool, default = True)
+
+            if selectNewProject:
+                project_utils.selectProject(project.id)
+                successEcho(f"Project \"{project.name}\" successfully selected.")
+
+    except EntityNotCreated:
+        raise click.ClickException(f"Failed to create project \"{name}\".")
+
+
+@click.command()
+@click.option("--project", "-p", type = str, help = "Project name")
+@click.option("--description", "-d", type = str, help = "Project description")
+def edit(name: Optional[str], description: Optional[str], visibility: Optional[int]) -> None:
     config = loadConfig()
-    config["projectId"] = projectId
-    saveConfig(config)
+    defaultProjectId = config.get("projectId")
+    if defaultProjectId is None and name is None:
+        errorEcho(f"To use edit command you need to specifiy project name using \"--project\" or \"-p\" flag, or you can select default project using \"coretex project select\" command.")
+
+    defaultProject = Project.fetchById(config["projectId"])
+
+    if name is None:
+        selectedProject = defaultProject
```

Comment:
You can assign the value of `defaultProject` variable directly, there is no need for `defaultProject` variable to exist